confgenerator/filter/internal/ast/ast.go

```diff
@@ -0,0 +1,432 @@
+	switch a := a.(type) {
+	case Expression:
+		return a.Simplify(), nil
+	}
+	return nil, fmt.Errorf("expected expression: %v", a)
+}
+
+// Conjunction represents an AND expression
+type Conjunction []Expression
+
+// Disjunction represents an OR expression
+type Disjunction []Expression
+
+func NewConjunction(a Attrib) (Conjunction, error) {
+	switch a := a.(type) {
+	case Conjunction:
+		return a, nil
+	case Expression:
+		return Conjunction{a.Simplify()}, nil
+	}
+	return nil, fmt.Errorf("expected expression: %v", a)
+}
+
+func (c Conjunction) Simplify() Expression {
+	if len(c) == 1 {
+		return c[0]
+	}
+	return c
+}
+
+func (c Conjunction) Append(a Attrib) (Conjunction, error) {
+	switch a := a.(type) {
+	case Conjunction:
+		return append(c, a...), nil
+	case Expression:
+		return append(c, a.Simplify()), nil
+	}
+	return nil, fmt.Errorf("expected expression: %v", a)
+}
+
+func (c Conjunction) Components(tag, key string) []fluentbit.Component {
+	var components []fluentbit.Component
+	m := modify(tag, key)
+	for i, e := range c {
+		subkey := fmt.Sprintf("%s_%d", key, i)
+		components = append(components, e.Components(tag, subkey)...)
+		m.OrderedConfig = append(m.OrderedConfig, [2]string{
+			"Condition", cond("Key_exists", subkey),
+		})
+	}
+	components = append(components, m)
+	return components
+}
+
+func NewDisjunction(a Attrib) (Disjunction, error) {
+	switch a := a.(type) {
+	case Disjunction:
+		return a, nil
+	case Expression:
+		return Disjunction{a.Simplify()}, nil
+	}
+	return nil, fmt.Errorf("expected expression: %v", a)
+}
+
+func (d Disjunction) Simplify() Expression {
+	if len(d) == 1 {
+		return d[0]
+	}
+	return d
+}
+
+func (d Disjunction) Append(a Attrib) (Disjunction, error) {
+	switch a := a.(type) {
+	case Disjunction:
+		return append(d, a...), nil
+	case Expression:
+		return append(d, a.Simplify()), nil
+	}
+	return nil, fmt.Errorf("expected expression: %v", a)
+}
+
+func (d Disjunction) Components(tag, key string) []fluentbit.Component {
+	var components []fluentbit.Component
+	var subkeys []string
+	for i, e := range d {
+		subkey := fmt.Sprintf("%s_%d", key, i)
+		components = append(components, e.Components(tag, subkey)...)
+		subkeys = append(subkeys, subkey)
+	}
+	// NB: We can't just pass key to e.Components because nested expressions might collide.
+	for _, subkey := range subkeys {
+		m := modify(tag, key)
+		m.Config["Condition"] = cond("Key_exists", subkey)
+		components = append(components, m)
+	}
+	return components
+}
+
+type Negation struct {
+	Expression
+}
+
+func (n Negation) Simplify() Expression {
+	return Negation{n.Expression.Simplify()}
+}
+
+func (n Negation) Components(tag, key string) []fluentbit.Component {
+	subkey := fmt.Sprintf("%s_0", key)
+	components := n.Expression.Components(tag, subkey)
+	m := modify(tag, key)
+	m.Config["Condition"] = cond("Key_does_not_exist", subkey)
+	components = append(components, m)
+	return components
+}
+
+func Unquote(in string) (string, error) {
```

Comment:
Nit: Add a comment with a link to the grammar and an explanation of why we can't just use `strconv.Unquote`

Reply:
Added a comment, hopefully I remembered the justification correctly.